### PR TITLE
Fix undefined reference in older browsers in useMessageBarReflow

### LINKED ISSUE #32237: [Bug]: useMessageBarReflow throws a TypeError on older browsers
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Linux 6.5 Ubuntu 22.04.4 LTS 22.04.4 LTS (Jammy Jellyfish)
    CPU: (32) x64 AMD EPYC 7763 64-Core Processor
    Memory: 111.20 GB / 125.78 GB
    Container: Yes
    Shell: 5.8.1 - /bin/zsh
```

### Are you reporting an Accessibility issue?

None

### Reproduction

Reproduction requires an older browser

## Actual Behavior
On older browsers (eg., Chrome 80), `borderBoxSize` is not supported. `useMessageBarReflow` makes an unguarded dereference of that value, causing a TypeError in such browsers

## Expected Behavior
No error should be thrown, and the component behavior should degrade gracefully.

### Logs

```shell
Error stack from Outlook telemetry on Chrome 80:

TypeError: Cannot read property '0' of undefined
    at ResizeObserver.<anonymous> (https://res.public.onecdn.static.microsoft/owamail/hashed-v1/scripts/owa.67643.m.5c33ac97.js:1:3225)
```

Translated:
```
at webpack://owa/node_modules/@fluentui/react-message-bar/lib/components/MessageBar/useMessageBarReflow.js:24
```

### Requested priority

High

### Products/sites affected

Outlook

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by MLoughry**
## Previous Behavior

In older browsers, such as Chrome 80, `ResizeObserverEntry.borderBoxSize` is not implemented, and returns undefined. The unguarded dereference in `useMessageBarReflow` causes a TypeError in such cases.

## New Behavior

Added an optional chaining guard `?.` to the code, and a comment explaining the reason for it.

## Related Issue(s)

- Fixes #32237

### CODE REVIEW

File: packages/react-components/react-message-bar/library/src/components/MessageBar/useMessageBarReflow.ts [hunk 1]
```diff
@@ -27,7 +27,8 @@ export function useMessageBarReflow(enabled: boolean = false) {
       }
 
       const entry = entries[0];
-      const borderBoxSize = entry?.borderBoxSize[0];
+      // `borderBoxSize` is not supported before Chrome 84, Firefox 92, nor Safari 15.4
```
Comment (spmonahan) [inline packages/react-components/react-message-bar/library/src/components/MessageBar/useMessageBarReflow.ts:30]: For Fluent's [support matrix](https://react.fluentui.dev/?path=/docs/concepts-developer-browser-support-matrix--page) only Safari is relevant.

File: packages/react-components/react-message-bar/library/src/components/MessageBar/useMessageBarReflow.ts [hunk 2]
```diff
@@ -27,7 +27,8 @@ export function useMessageBarReflow(enabled: boolean = false) {
...
+      const borderBoxSize = entry?.borderBoxSize?.[0];
```
Comment (spmonahan) [inline packages/react-components/react-message-bar/library/src/components/MessageBar/useMessageBarReflow.ts:31]: I think we can expand support by using `getBoundingClientRect()` when `borderBoxSize` is unavailable.

https://github.com/microsoft/fluentui-contrib/blob/main/packages/houdini-utils/src/paint/fallback/anims/play.ts#L28-L35

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports | Baseline (minified/GZIP) | PR    | Change     |
| :---------------- | -----------------------: | ----: | ---------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ReactComponents.fixture.js'>react-components: entire library</abbr> | `1.095 MB`<br />`270.595 kB` | `1.095 MB`<br />`270.607 kB` | `32 B` <br />`12 B` |
| <samp>react-message-bar</samp> <br /> <abbr title='bundle-size/Default.fixture.js'>MessageBar (all components)</abbr> | `24.596 kB`<br />`9.18 kB` | `24.628 kB`<br />`9.196 kB` | `32 B` <br />`16 B` |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports | Size (minified/GZIP) |
| ----------------- | -------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> | `69.141 kB`<br />`20.157 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `217.412 kB`<br />`63.063 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr> | `44.442 kB`<br />`14.607 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr> | `8.39 kB`<br />`2.64 kB` |
| <samp>react-timepicker-compat</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr> | `106.751 kB`<br />`35.596 kB` |
</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/ba1dcc42dad00aad42b5942718da22bfb7d9fcac'>ba1dcc42dad00aad42b5942718da22bfb7d9fcac</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/Avatar-mount_base.html">655</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/Avatar-mount.html">639</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/Button-mount_base.html">317</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/Button-mount.html">307</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/Field-mount_base.html">1148</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/Field-mount.html">1126</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/FluentProvider-mount_base.html">715</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/FluentProvider-mount.html">707</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/FluentProviderWithTheme-mount_base.html">83</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/FluentProviderWithTheme-mount.html">79</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">30</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">35</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">75</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">73</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/MakeStyles-mount_base.html">873</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/MakeStyles-mount.html">857</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/Persona-mount_base.html">1775</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/Persona-mount.html">1714</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/SpinButton-mount_base.html">1394</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/SpinButton-mount.html">1446</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/SwatchPicker-mount_base.html">1692</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/32238/perf-test-react-components/SwatchPicker-mount.html">1667</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-message-bar/library/src/components/MessageBar/useMessageBarReflow.ts (hunk 1 above)
Comment (MLoughry) [reply packages/react-components/react-message-bar/library/src/components/MessageBar/useMessageBarReflow.ts:30]: As mentioned in chat, that matrix doesn't align with Outlook or M365's support matrix.

**Review by ling1726 — APPROVED** [review]

File: change/@fluentui-react-message-bar-bfb5f0c1-b637-4cbd-96e4-9ea0283c03d2.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-message-bar-bfb5f0c1-b637-4cbd-96e4-9ea0283c03d2.json:1]: ## 🕵🏾‍♀️ [<b data-uid='vrt-gist'></b> visual regressions to review in the fluentuiv9 Visual Regression Report](https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=32238&commitId=c17afed77d461f3a2ddf6cbaf26e5bb0d9087a2a&env=prod&clientType=FLUENTUI)


  <details><summary><b>Card Converged - Interactive</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Card Converged - Interactive.appearance interactive - Filled Alternative.click.chromium.png|0| Removed|
|Card Converged - Interactive.appearance interactive - Filled Alternative.hover.chromium.png|0| Removed|
|Card Converged - Interactive.appearance interactive - Filled Alternative.normal.chromium.png|0| Removed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Pull request merged**